suppressible-error-prone/src/main/java/com/palantir/suppressibleerrorprone/LazySuppressionReplacement.java

```diff
@@ -27,42 +27,37 @@
 import java.util.Set;
 import java.util.stream.Collectors;
 
-final class SuppressingReplacement extends Replacement {
-    // We really do need to be this lazy for generating the Replacements, as error-prone immediately converts
-    // the Fix to a Replacement when a Description is given to it, and we need to defer the computation of the
-    // Replacement until a number of Descriptions have been produced, to handle multiple errors being suppressed
-    // at the same level.
-
+/**
+ * A Replacement that handles @SuppressWarnings modifications with line-removing capabilities.
+ * When the final replacement text is empty (no suppressions), it will remove preceding
+ * whitespace up to and including the newline.
+ */
+final class LazySuppressionReplacement extends Replacement {
     private final Range<Integer> range;
     private final List<String> existingSuppressions;
     private final String suffix;
-    private final Set<String> newSuppressions;
+    private final Set<String> desiredSuppressions;
```

Comment:
This implicitly uses a listener/accumulator pattern, like error-prone's DescriptionListener. Maybe make this explicit